thoth-app/src/component/publications_form.rs

```diff
@@ -282,6 +307,48 @@ impl Component for PublicationsFormComponent {
                     false
                 }
             }
+            Msg::ChangeWeightG(value) => {
```

Comment:
This could be flattened a bit more for clarity, e.g.:

```
Msg::ChangeWeightG(value) => {
    let changed_value = self.new_publication.weight_g.neq_assign(value.to_opt_float());
    if self.convert_weights && changed_value {
        // Automatically update paired weight field with default conversion.
        let mut weight_oz = None;
        if let Some(weight_g) = self.new_publication.weight_g {
            weight_oz = Some(weight_g.convert_weight_from_to(&WeightUnit::G, &WeightUnit::Oz));
        }
        self.new_publication.weight_oz.neq_assign(weight_oz);                 
    }
    changed_value
```